### DISCUSSION TITLE
Getting issue while setting up swc

### DISCUSSION THREAD

**Discussion by sheikhfahad67** [swc-project/swc#7849] [Q&A] [answered]
I followed the documentation but got the issue after following steps:
**Steps:**
1. npm i --save-dev @swc/cli @swc/core
2. Change compilerOption - builder : swc
3. Create .swcrc file with provided configuration

**Issue:**
"In other works, SWC does not know which directory should be used as a base directory. It can be deduced if `.swcrc` is used, but if not, there are many candidates. e.g. the directory containing `package.json`, or the current working directory. Because of that, the caller (typically the developer of the JavaScript package) should specify it. If you see this error, please report an issue to the package author. 
Failed to compile 5 files with swc."

**Comment by kdy1** [comment, marked as answer]
You need to configure SWC correctly, and I can't know anything if you don't provide the config

**Reply by kdy1** [reply]
The error message is pretty accurate, though. Please read it.

**Reply by sheikhfahad67** [reply]
Thank you for your response:

I added this configuration in .swcrc file

```
{
  "$schema": "https://json.schemastore.org/swcrc",
  "sourceMaps": true,
  "jsc": {
    "parser": {
      "syntax": "typescript",
      "decorators": true,
      "dynamicImport": true
    },
    "baseUrl": "./"
  },
  "minify": false
}
```
I also try to chnage the baseUrl to "/src" but after that getting different issue

**Reply by sheikhfahad67** [reply]
Here is the nest-cli.json

```
{
  "$schema": "https://json.schemastore.org/nest-cli",
  "collection": "@nestjs/schematics",
  "sourceRoot": "src",
  "compilerOptions": {
    "deleteOutDir": true,
    "builder": "swc",
    "typeCheck": true
  }
}
```

**Reply by kdy1** [reply]
Ah, you are using nestjs.
nestjs has a bug related to `jsc.paths`

**Reply by sheikhfahad67** [reply]
Is there any workaround? I followed the same steps as mentioned in the official documentation of the Nestjs 
link here: https://docs.nestjs.com/recipes/swc

**Reply by kdy1** [reply]
Can you try `v10.1.15`? https://github.com/nestjs/nest-cli/commit/deddb98e545846e44e5c5d3c2585162365dfbf18 should fix it.

**Reply by sheikhfahad67** [reply]
Thank you, I'll try it.
